[Alvin Lim] iP

alvinlim277 opened the pull request:
# Dook
### A task manager application
> This is a quote.

Dook helps you keep track of your tasks!
- text based
- easy to learn
- ~~Fast~~ SUPER FAST to use

### Instructions for use:
1. Download from [here](https://github.com/alvinlim277/ip/releases).
2. Run the application
3. Add your tasks
4. Let Dook add some order to your mess of a life 💯 

Also, it's all **FOR FREE**

### Some Features
- [X] Managing todos, deadlines and events
- [X] Querying by name and by date
- [ ] Reminders

Here is some code for no reason at all.
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

alvinlim277 committed:
Add greet and exit

alvinlim277 committed:
Add echo

alvinlim277 committed:
Add list

alvinlim277 committed:
Add marking/unmarking

alvinlim277 committed:
Add event, todo, deadline

alvinlim277 committed:
Add semi-automated testing

alvinlim277 committed:
Add exception handling and error messages

alvinlim277 committed:
Add delete functionality

alvinlim277 committed:
Add enums and fix grammar

alvinlim277 committed:
Add user guide on invalid command

alvinlim277 committed:
Add DookList to separate responsibilities

alvinlim277 committed:
Update expected output

alvinlim277 committed:
Separate responsibilities and start adding save.

alvinlim277 committed:
Add save functionality

alvinlim277 committed:
Merge branch 'branch-level-7'

alvinlim277 committed:
Add date time recognition

alvinlim277 committed:
Merge branch 'branch-level-8'

alvinlim277 committed:
Add more OOP and command pattern

alvinlim277 committed:
Merge branch 'branch-more-oop'

alvinlim277 committed:
Divide classes into packages

alvinlim277 committed:
Remove old classes

alvinlim277 committed:
Merge branch 'branch-more-oop' into branch-packages

alvinlim277 committed:
Merge branch 'branch-packages'

alvinlim277 committed:
Merge branch 'add-gradle-support' of https://github.com/alvinlim277/ip

alvinlim277 committed:
Add gradle support and checkstyle

alvinlim277 committed:
Merge branch 'branch-gradle'

alvinlim277 committed:
Start adding tests

alvinlim277 committed:
Add tests for TaskList and Deadline

alvinlim277 committed:
Merge branch 'branch-junit'

alvinlim277 committed:
Fix file reading bugs

alvinlim277 committed:
Merge branch 'branch-jar'

alvinlim277 committed:
Add JavaDocs for public methods

alvinlim277 committed:
Fix coding standard

alvinlim277 committed:
Add find functionality

alvinlim277 committed:
Merge branch 'branch-A-CodingStandard'

alvinlim277 committed:
Merge branch 'branch-A-JavaDoc'

alvinlim277 committed:
Merge branch 'branch-Level-9'

alvinlim277 committed:
Add missing JavaDocs and fix style

alvinlim277 committed:
Remove redundant methods

alvinlim277 committed:
Add accumulator function

ivanleekk commented on src/main/java/dook/services/UiDisplay.java:
```diff
@@ -0,0 +1,64 @@
+/**
+ * Responsible for displaying messages to the user, as well as reading user commands.
+ */
+public class UiDisplay {
```
Good job making sure UI is written as Ui for the class name, as abbreviations should not be in upper case when part of a name

ivanleekk commented on src/main/java/dook/services/Parser.java:
```diff
@@ -0,0 +1,177 @@
+            return new InvalidCommand();
+        }
+    }
+    private Command handleMark(String body, boolean value) throws DookException {
```
Good job on following the convention of starting these methods which do things as a verb.

ivanleekk commented on src/main/java/dook/task/Event.java:
```diff
@@ -0,0 +1,78 @@
+        }
+    }
+    @Override
+    public boolean isDuring(LocalDate dateTime) {
```
good use of prefixes like is, has or was to denote booleans, keep it up :)

alvinlim277 replied:
Thanks for all of the feedback!

shashahchk commented on src/main/java/dook/command/AddTaskCommand.java:
```diff
@@ -0,0 +1,31 @@
+ */
+public class AddTaskCommand extends Command {
+    private final Task task;
+    public AddTaskCommand(Task task) {
```
perhaps can add a newline between the field and the method for better readability? I noticed the same issue in several other places too

alvinlim277 replied:
Definitely agree, thanks for the suggestion!

shashahchk commented on src/main/java/dook/command/InvalidCommand.java:
```diff
@@ -0,0 +1,25 @@
+/**
+ * Represents any invalid command.
+ */
+public class InvalidCommand extends Command {
```
good naming and structuring of classes

shashahchk commented on src/main/java/dook/command/Command.java:
```diff
@@ -0,0 +1,31 @@
+/**
+ * Abstract class used for the Command pattern.
+ */
+public abstract class Command {
```
I like how you make command an abstract class!

shashahchk commented on src/main/java/dook/task/Deadline.java:
```diff
@@ -0,0 +1,69 @@
+ * Task with a deadline.
+ */
+public class Deadline extends Task implements TimedTask {
+    protected String by;
```
perhaps can make the naming of this field a little more descriptive?

alvinlim277 replied:
Will look into it, thanks for all the feedback!

shashahchk commented on src/main/java/dook/services/Parser.java:
```diff
@@ -0,0 +1,177 @@
+import java.time.format.DateTimeParseException;
+
+import dook.DookException;
+import dook.command.*;
```
Should this wildcard import be changed to separate imports?

alvinlim277 replied:
Good catch, will change this.

shashahchk commented on src/main/java/dook/services/Parser.java:
```diff
@@ -0,0 +1,177 @@
+        return new AfterCommand(localDate);
+    }
+
+    private Command handleDuring(String body) throws DookException {
```
I like how you name these methods with the same prefix to handle different cases.

shashahchk reviewed (commented):
LGTM. A few nits to fix.

alvinlim277 committed:
Add GUI

alvinlim277 committed:
Fix all checkstyle violations and improve on PR feedback

alvinlim277 committed:
Merge branch 'branch-minor-fixes'

DistractedCat commented on src/main/java/dook/Dook.java:
```diff
@@ -0,0 +1,52 @@
+ *  Class containing the main execution function.
+ */
+public class Dook {
+    public static final Path PROPER_PATH = Paths.get("dook.txt");
```
Good naming for constant and final variables

DistractedCat commented on src/main/java/dook/command/BeforeCommand.java:
```diff
@@ -0,0 +1,39 @@
+     * @throws DookException Exception thrown by Dook.
+     */
+    @Override
+    public String execute(Storage storage, TaskList taskList) throws DookException {
```
Good choice of names for functions, using verbs to describe the action!

DistractedCat commented on src/main/java/dook/command/Command.java:
```diff
@@ -0,0 +1,26 @@
+ */
+public abstract class Command {
+    protected boolean isExit = false;
+    public boolean getIsExit() {
```
```suggestion
    public boolean isExit() {
```

DistractedCat reviewed (commented):
LGTM!

alvinlim277 committed:
Improve GUI

alvinlim277 committed:
Merge branch 'branch-minor-fixes'

alvinlim277 committed:
Add assertions at key points in code.

Code does not make use of assertions to test assumptions about the program.
This needs to change as assertions can be used liberally without affecting performance drastically, and will highlight any programmer errors in the code.

Let's,
* Add some assertions at key points in the Parser, Storage and TaskList classes.

alvinlim277 committed:
Improve code quality.

There are many instances in the code where code quality concentions
are not adhered to, hindering readability and ease of understanding.

Let's,
* Apply SLAP Principle.
* Replace 'magic strings' with named constants.

alvinlim277 committed:
Merge pull request #1 from alvinlim277/branch-A-Assertions

Add assertions at key points in code.

alvinlim277 committed:
Merge branch 'master' into branch-A-CodeQuality

alvinlim277 committed:
Merge pull request #2 from alvinlim277/branch-A-CodeQuality

Improve code quality.

alvinlim277 committed:
Create gradle.yml

alvinlim277 committed:
Implement user-defined command syntax

Users have to type commands fully when they might prefer defining
their own set of shorter command keywords.

Let's,
* Let users save their preferred commands in a text file.
* Let users map multiple keywords to the same command.

alvinlim277 committed:
Merge branch 'branch-C-FriendlierSyntax'

alvinlim277 committed:
Update README.md

alvinlim277 committed:
Fix ByeCommand bug

alvinlim277 committed:
Fix header comments

alvinlim277 committed:
Merge branch 'branch-final-submission'
